eva_submission/eload_deletion.py

```diff
@@ -0,0 +1,118 @@
+        shutil.rmtree(archive_dir, ignore_errors=True)
+        os.makedirs(archive_dir, exist_ok=True)
+
+        # copy relevant files to the archive_dir
+        self.copy_eload_files(archive_dir)
+
+        # gzip each file if they are not already compressed
+        for root, _, files in os.walk(archive_dir):
+            for file in files:
+                file_path = os.path.join(root, file)
+                if not self.is_compressed(file):
+                    gzip_path = f"{file_path}.gz"
+                    with open(file_path, 'rb') as f_in, gzip.open(gzip_path, 'wb') as f_out:
+                        shutil.copyfileobj(f_in, f_out)
+                    os.remove(file_path)
+
+        # Create a tar archive of the entire archive_dir
+        archive_tar_file = os.path.join(self.eload_dir, f'{self.eload}.tar')
+        with tarfile.open(archive_tar_file, mode="w") as tar:
+            for root, _, files in os.walk(archive_dir):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    arcname = os.path.relpath(file_path, start=archive_dir)  # Avoid nesting archive_dir
+                    tar.add(file_path, arcname=arcname)
+
+        # copy to LTS
+        try:
+            shutil.copy(archive_tar_file, self.lts_archive_file)
+        except Exception as e:
+            print(f"Error copying archive to LTS: {e}")
+
+    def copy_eload_files(self, archive_dir):
+        # copy config file
+        shutil.copy(self.config_path, archive_dir)
+
+        # copy submission logs
+        for file in Path(self.eload_dir).glob("*_submission.log"):
+            shutil.copy(file, archive_dir)
+
+        # copy metadata spreadsheet and vcf files along with index
+        src_ena_dir = os.path.join(self.eload_dir, '18_brokering/ena')
+        archive_ena_dir = os.path.join(archive_dir, '18_brokering/ena')
+        os.makedirs(archive_ena_dir, exist_ok=True)
+        shutil.copy(os.path.join(src_ena_dir, 'metadata_spreadsheet.xlsx'), archive_ena_dir)
+        for file in Path(src_ena_dir).glob("*.vcf.gz"):
+            shutil.copy(file, archive_ena_dir)
+        for file in Path(src_ena_dir).glob("*.vcf.csi"):
+            shutil.copy(file, archive_ena_dir)
+
+        # copy 00_logs
+        src_log_dir = os.path.join(self.eload_dir, '00_logs')
+        real_src_log_dir = os.path.realpath(src_log_dir)
+        archive_log_dir = os.path.join(archive_dir, '00_logs')
+        shutil.copytree(real_src_log_dir, archive_log_dir)
+
+        # copy accessioned files from 60_eva_public
+        src_accessioned_files_dir = os.path.join(self.eload_dir, "60_eva_public")
+        real_accessioned_files_dir = os.path.realpath(src_accessioned_files_dir)
+        archive_accession_files_dir = os.path.join(archive_dir, "60_eva_public")
+        os.makedirs(archive_accession_files_dir, exist_ok=True)
+        for file in Path(real_accessioned_files_dir).glob("*.accessioned.vcf.gz"):
```

Comment:
Should we also keep the index file `accessioned.vcf.gz.csi`?

Reply:
updated to copy accessioned index files